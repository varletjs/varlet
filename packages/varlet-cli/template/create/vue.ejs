<template>
  <div class="var-<%- kebabCaseName %>"></div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import './<%- kebabCaseName %>.less'

export default defineComponent({
  name: '<%- namespace + bigCamelizeName %>',
})
</script>
