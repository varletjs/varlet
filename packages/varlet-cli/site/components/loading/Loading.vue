<template>
  <div :class="n()">
    <div :class="classes(n('content'), [loading, n('content--active')])" v-if="$slots.default">
      <slot />
      <div :class="n('content-mask')" v-if="loading"></div>
    </div>
    <div :class="classes('var-site--box', n('body'), [$slots.default, n('inside')])" v-if="isShow">
      <div :class="n('circle')" v-if="type === 'circle'">
        <span
          :class="classes(n('circle-block'), n(`circle-block--${size}`))"
          :style="{
            width: multiplySizeUnit(radius, 2),
            height: multiplySizeUnit(radius, 2),
            color,
          }"
        >
          <svg viewBox="25 25 50 50">
            <circle cx="50" cy="50" r="20" fill="none"></circle>
          </svg>
        </span>
      </div>

      <template v-for="(nums, key) in loadingTypeDict" :key="key">
        <div :class="classes(n(key), n(`${key}--${size}`))" v-if="type === key">
          <div
            v-for="num in nums"
            :key="num + key"
            :style="{ backgroundColor: color }"
            :class="classes(n(`${key}-item`), n(`${key}-item--${size}`))"
          ></div>
        </div>
      </template>
      <div
        :class="classes(n('description'), n(`description--${size}`))"
        :style="{ color }"
        v-if="$slots.description || description"
      >
        <slot name="description">{{ description }}</slot>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import { computed, defineComponent } from 'vue'
import { props } from './props'
import { createNamespace, call } from '../utils/components'
import { multiplySizeUnit } from '../utils/elements'
import type { ComputedRef } from 'vue'

const { n, classes } = createNamespace('loading')

export default defineComponent({
  name: 'VarLoading',
  props,
  setup(props, { slots }) {
    const loadingTypeDict = {
      wave: 5,
      cube: 4,
      rect: 8,
      disappear: 3,
    }

    const isShow: ComputedRef<boolean> = computed(() => {
      if (!call(slots.default)) return true

      return props.loading
    })

    return {
      n,
      classes,
      multiplySizeUnit,
      loadingTypeDict,
      isShow,
    }
  },
})
</script>

<style lang="less">
@import '../styles/common';
@import './loading';
</style>
