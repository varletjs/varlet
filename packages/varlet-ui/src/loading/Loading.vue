<template>
  <div :class="n()">
    <div :class="classes(n('content'), [loading, n('content--active')])" v-if="$slots.default">
      <slot />
    </div>
    <div :class="classes('var--box', n('body'), [$slots.default, n('inside')])" v-if="isShow">
      <div :class="n('circle')" v-if="type === 'circle'">
        <span
          :class="n('circle-block')"
          :style="{
            width: getRadius * 2 + 'px',
            height: getRadius * 2 + 'px',
            color,
          }"
        >
          <svg viewBox="25 25 50 50">
            <circle cx="50" cy="50" r="20" fill="none"></circle>
          </svg>
        </span>
      </div>

      <template v-for="(nums, key) in loadingTypeDict" :key="key">
        <div :class="classes(n(key), n(`${key}--${size}`))" v-if="type === key">
          <div
            v-for="num in nums"
            :key="num + key"
            :style="{ backgroundColor: color }"
            :class="classes(n(`${key}-item`), n(`${key}-item--${size}`))"
          ></div>
        </div>
      </template>
      <div
        :class="classes(n('description'), n(`description--${size}`))"
        :style="{ color }"
        v-if="$slots.description || description"
      >
        <slot name="description">{{ description }}</slot>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import { computed, defineComponent } from 'vue'
import { props } from './props'
import { toNumber } from '../utils/shared'
import type { ComputedRef } from 'vue'
import { createNamespace, call } from '../utils/components'

const { n, classes } = createNamespace('loading')

export default defineComponent({
  name: 'VarLoading',
  props,
  setup(props, { slots }) {
    const loadingTypeDict = {
      wave: 5,
      cube: 4,
      rect: 8,
      disappear: 3,
    }

    const sizeDict = {
      mini: 9,
      small: 12,
      normal: 15,
      large: 18,
    }

    const getRadius: ComputedRef<number> = computed(() => {
      return props.radius ? toNumber(props.radius) : sizeDict[props.size]
    })

    const isShow: ComputedRef<boolean> = computed(() => {
      if (!call(slots.default)) return true

      return props.loading
    })

    return {
      n,
      classes,
      loadingTypeDict,
      getRadius,
      isShow,
    }
  },
})
</script>

<style lang="less">
@import '../styles/common';
@import './loading';
</style>
