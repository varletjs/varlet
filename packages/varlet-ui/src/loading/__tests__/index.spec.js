import Loading from '..'
import VarLoading from '../Loading'
import { mount } from '@vue/test-utils'
import { createApp } from 'vue'

test('test loading plugin', () => {
  const app = createApp({}).use(Loading)
  expect(app.component(Loading.name)).toBeTruthy()
})

describe('test loading component props', () => {
  test('test loading color', () => {
    const wrapper = mount(VarLoading, {
      props: { color: 'red' },
    })

    expect(wrapper.find('.var-loading__circle-block').attributes('style')).toContain('color: red;')
    wrapper.unmount()
  })

  test('test loading type', () => {
    ;['circle', 'wave', 'cube', 'rect', 'disappear'].forEach((type) => {
      const wrapper = mount(VarLoading, {
        props: { type },
      })

      expect(wrapper.find('.var-loading__' + type).exists()).toBe(true)
      wrapper.unmount()
    })
  })

  test('test loading size', () => {
    ;['large', 'normal', 'small', 'mini'].forEach((size) => {
      const wrapper = mount(VarLoading, {
        props: { size },
      })

      expect(wrapper.find('.var-loading__circle-block').classes()).toContain(`var-loading__circle-block--${size}`)
      wrapper.unmount()
    })
  })

  test('test loading description', () => {
    const wrapper = mount(VarLoading, {
      props: {
        description: 'This is description',
      },
    })

    expect(wrapper.find('.var-loading__description').element.textContent).toBe('This is description')
    wrapper.unmount()
  })

  test('test loading loading', async () => {
    const wrapper = mount(VarLoading, {
      props: { loading: true },
      slots: {
        default: () => 'This is default slots',
      },
    })

    expect(wrapper.find('.var-loading__content-mask').exists()).toBe(true)
    await wrapper.setProps({ loading: false })
    expect(wrapper.find('.var-loading__content-mask').exists()).toBe(false)
    wrapper.unmount()
  })

  test('test loading radius', () => {
    ;['circle', 'wave', 'cube', 'rect', 'disappear'].forEach(async (type) => {
      const wrapper = mount(VarLoading, {
        props: {
          radius: '20px',
          type,
        },
      })

      type === 'circle'
        ? expect(wrapper.find('.var-loading__circle-block').attributes('style')).toContain('width: 40px; height: 40px;')
        : expect(wrapper.find('.var-loading__circle-block').exists()).toBe(false)
      await wrapper.setProps({ radius: 30 })
      type === 'circle'
        ? expect(wrapper.find('.var-loading__circle-block').attributes('style')).toContain('width: 60px; height: 60px;')
        : expect(wrapper.find('.var-loading__circle-block').exists()).toBe(false)
      wrapper.unmount()
    })
  })
})

describe('test loading component slots', () => {
  test('test loading default slots', () => {
    const wrapper = mount(VarLoading, {
      slots: {
        default: () => 'This is default slots',
      },
    })

    expect(wrapper.find('.var-loading__content').element.textContent).toBe('This is default slots')
    wrapper.unmount()
  })

  test('test loading description slots', () => {
    const wrapper = mount(VarLoading, {
      slots: {
        description: () => 'This is description slots',
      },
    })

    expect(wrapper.find('.var-loading__description').element.textContent).toBe('This is description slots')
    wrapper.unmount()
  })
})
