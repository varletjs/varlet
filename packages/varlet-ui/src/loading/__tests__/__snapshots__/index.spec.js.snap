// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test circle props 1`] = `
"<div class=\\"var-loading\\">
  <!--v-if-->
  <div class=\\"var--box var-loading__body\\">
    <div class=\\"var-loading__circle\\"><span class=\\"var-loading__circle-block\\" style=\\"width: 24px; height: 24px; color: rgb(41, 121, 255);\\"><svg viewBox=\\"25 25 50 50\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\"></circle></svg></span></div>
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`test cube props 1`] = `
"<div class=\\"var-loading\\">
  <!--v-if-->
  <div class=\\"var--box var-loading__body\\">
    <!--v-if-->
    <!--v-if-->
    <div class=\\"var-loading__cube var-loading__cube--small\\">
      <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__cube-item var-loading__cube-item--small\\"></div>
      <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__cube-item var-loading__cube-item--small\\"></div>
      <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__cube-item var-loading__cube-item--small\\"></div>
      <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__cube-item var-loading__cube-item--small\\"></div>
    </div>
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`test disappear props 1`] = `
"<div class=\\"var-loading\\">
  <!--v-if-->
  <div class=\\"var--box var-loading__body\\">
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
    <div class=\\"var-loading__disappear var-loading__disappear--small\\">
      <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__disappear-item var-loading__disappear-item--small\\"></div>
      <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__disappear-item var-loading__disappear-item--small\\"></div>
      <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__disappear-item var-loading__disappear-item--small\\"></div>
    </div>
    <!--v-if-->
  </div>
</div>"
`;

exports[`test loading example 1`] = `
"<div class=\\"loading-demo\\">
  <div class=\\"loading-demo__type\\">
    <div class=\\"app-type\\">类型</div>
    <div class=\\"loading-demo__type-block\\">
      <div class=\\"var-loading\\">
        <!--v-if-->
        <div class=\\"var--box var-loading__body\\">
          <div class=\\"var-loading__circle\\"><span class=\\"var-loading__circle-block\\" style=\\"width: 30px; height: 30px; color: currentColor;\\"><svg viewBox=\\"25 25 50 50\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\"></circle></svg></span></div>
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
      <div class=\\"var-loading\\">
        <!--v-if-->
        <div class=\\"var--box var-loading__body\\">
          <!--v-if-->
          <!--v-if-->
          <div class=\\"var-loading__cube var-loading__cube--normal\\">
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__cube-item var-loading__cube-item--normal\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__cube-item var-loading__cube-item--normal\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__cube-item var-loading__cube-item--normal\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__cube-item var-loading__cube-item--normal\\"></div>
          </div>
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
      <div class=\\"var-loading\\">
        <!--v-if-->
        <div class=\\"var--box var-loading__body\\">
          <!--v-if-->
          <div class=\\"var-loading__wave var-loading__wave--normal\\">
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__wave-item var-loading__wave-item--normal\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__wave-item var-loading__wave-item--normal\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__wave-item var-loading__wave-item--normal\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__wave-item var-loading__wave-item--normal\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__wave-item var-loading__wave-item--normal\\"></div>
          </div>
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
      <div class=\\"var-loading\\">
        <!--v-if-->
        <div class=\\"var--box var-loading__body\\">
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <div class=\\"var-loading__rect var-loading__rect--normal\\">
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item--normal\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item--normal\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item--normal\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item--normal\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item--normal\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item--normal\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item--normal\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item--normal\\"></div>
          </div>
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
      <div class=\\"var-loading\\">
        <!--v-if-->
        <div class=\\"var--box var-loading__body\\">
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <div class=\\"var-loading__disappear var-loading__disappear--normal\\">
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__disappear-item var-loading__disappear-item--normal\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__disappear-item var-loading__disappear-item--normal\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__disappear-item var-loading__disappear-item--normal\\"></div>
          </div>
          <!--v-if-->
        </div>
      </div>
    </div>
  </div>
  <div class=\\"loading-demo__type\\">
    <div class=\\"app-type\\">颜色</div>
    <div class=\\"loading-demo__type-block\\">
      <div class=\\"var-loading\\">
        <!--v-if-->
        <div class=\\"var--box var-loading__body\\">
          <div class=\\"var-loading__circle\\"><span class=\\"var-loading__circle-block\\" style=\\"width: 30px; height: 30px; color: rgb(41, 121, 255);\\"><svg viewBox=\\"25 25 50 50\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\"></circle></svg></span></div>
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
      <div class=\\"var-loading\\">
        <!--v-if-->
        <div class=\\"var--box var-loading__body\\">
          <!--v-if-->
          <!--v-if-->
          <div class=\\"var-loading__cube var-loading__cube--normal\\">
            <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__cube-item var-loading__cube-item--normal\\"></div>
            <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__cube-item var-loading__cube-item--normal\\"></div>
            <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__cube-item var-loading__cube-item--normal\\"></div>
            <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__cube-item var-loading__cube-item--normal\\"></div>
          </div>
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
      <div class=\\"var-loading\\">
        <!--v-if-->
        <div class=\\"var--box var-loading__body\\">
          <!--v-if-->
          <div class=\\"var-loading__wave var-loading__wave--normal\\">
            <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__wave-item var-loading__wave-item--normal\\"></div>
            <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__wave-item var-loading__wave-item--normal\\"></div>
            <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__wave-item var-loading__wave-item--normal\\"></div>
            <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__wave-item var-loading__wave-item--normal\\"></div>
            <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__wave-item var-loading__wave-item--normal\\"></div>
          </div>
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
      <div class=\\"var-loading\\">
        <!--v-if-->
        <div class=\\"var--box var-loading__body\\">
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <div class=\\"var-loading__rect var-loading__rect--normal\\">
            <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__rect-item var-loading__rect-item--normal\\"></div>
            <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__rect-item var-loading__rect-item--normal\\"></div>
            <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__rect-item var-loading__rect-item--normal\\"></div>
            <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__rect-item var-loading__rect-item--normal\\"></div>
            <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__rect-item var-loading__rect-item--normal\\"></div>
            <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__rect-item var-loading__rect-item--normal\\"></div>
            <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__rect-item var-loading__rect-item--normal\\"></div>
            <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__rect-item var-loading__rect-item--normal\\"></div>
          </div>
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
      <div class=\\"var-loading\\">
        <!--v-if-->
        <div class=\\"var--box var-loading__body\\">
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <div class=\\"var-loading__disappear var-loading__disappear--normal\\">
            <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__disappear-item var-loading__disappear-item--normal\\"></div>
            <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__disappear-item var-loading__disappear-item--normal\\"></div>
            <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__disappear-item var-loading__disappear-item--normal\\"></div>
          </div>
          <!--v-if-->
        </div>
      </div>
    </div>
  </div>
  <div class=\\"loading-demo__size\\">
    <div class=\\"app-type\\">大小</div>
    <div class=\\"loading-demo__size-block\\">
      <div class=\\"var-loading\\">
        <!--v-if-->
        <div class=\\"var--box var-loading__body\\">
          <div class=\\"var-loading__circle\\"><span class=\\"var-loading__circle-block\\" style=\\"width: 24px; height: 24px; color: currentColor;\\"><svg viewBox=\\"25 25 50 50\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\"></circle></svg></span></div>
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
      <div class=\\"var-loading\\">
        <!--v-if-->
        <div class=\\"var--box var-loading__body\\">
          <!--v-if-->
          <!--v-if-->
          <div class=\\"var-loading__cube var-loading__cube--small\\">
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__cube-item var-loading__cube-item--small\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__cube-item var-loading__cube-item--small\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__cube-item var-loading__cube-item--small\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__cube-item var-loading__cube-item--small\\"></div>
          </div>
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
      <div class=\\"var-loading\\">
        <!--v-if-->
        <div class=\\"var--box var-loading__body\\">
          <!--v-if-->
          <div class=\\"var-loading__wave var-loading__wave--small\\">
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__wave-item var-loading__wave-item--small\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__wave-item var-loading__wave-item--small\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__wave-item var-loading__wave-item--small\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__wave-item var-loading__wave-item--small\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__wave-item var-loading__wave-item--small\\"></div>
          </div>
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
      <div class=\\"var-loading\\">
        <!--v-if-->
        <div class=\\"var--box var-loading__body\\">
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <div class=\\"var-loading__rect var-loading__rect--small\\">
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item--small\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item--small\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item--small\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item--small\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item--small\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item--small\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item--small\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item--small\\"></div>
          </div>
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
      <div class=\\"var-loading\\">
        <!--v-if-->
        <div class=\\"var--box var-loading__body\\">
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <div class=\\"var-loading__disappear var-loading__disappear--small\\">
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__disappear-item var-loading__disappear-item--small\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__disappear-item var-loading__disappear-item--small\\"></div>
            <div style=\\"background-color: currentColor;\\" class=\\"var-loading__disappear-item var-loading__disappear-item--small\\"></div>
          </div>
          <!--v-if-->
        </div>
      </div>
    </div>
  </div>
  <div>
    <div class=\\"app-type\\">包裹内容</div><button class=\\"var-button var--box var-button--normal var--inline-flex var-button--default var-elevation--2\\" style=\\"margin-bottom: 8px;\\">
      <!--v-if-->
      <div class=\\"var-button__content\\">打开</div>
    </button>
    <div class=\\"var-loading\\">
      <div class=\\"var-loading__content\\">
        <div class=\\"var-card var-elevation--2\\">
          <!--v-if-->
          <div class=\\"var-card__title\\">介绍</div>
          <!--v-if-->
          <div class=\\"var-card__description\\">Varlet 是一个基于Vue3开发的 Material 风格移动端组件库，全面拥抱Vue3生态，由社区的小伙伴开发和维护。</div>
          <!--v-if-->
        </div>
      </div>
      <!--v-if-->
    </div>
  </div>
</div>"
`;

exports[`test rect props 1`] = `
"<div class=\\"var-loading\\">
  <!--v-if-->
  <div class=\\"var--box var-loading__body\\">
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
    <div class=\\"var-loading__rect var-loading__rect--small\\">
      <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__rect-item var-loading__rect-item--small\\"></div>
      <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__rect-item var-loading__rect-item--small\\"></div>
      <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__rect-item var-loading__rect-item--small\\"></div>
      <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__rect-item var-loading__rect-item--small\\"></div>
      <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__rect-item var-loading__rect-item--small\\"></div>
      <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__rect-item var-loading__rect-item--small\\"></div>
      <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__rect-item var-loading__rect-item--small\\"></div>
      <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__rect-item var-loading__rect-item--small\\"></div>
    </div>
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`test slots 1`] = `
"<div class=\\"var-loading\\">
  <div class=\\"var-loading__content\\">
    <div>test</div>
  </div>
  <!--v-if-->
</div>"
`;

exports[`test slots 2`] = `
"<div class=\\"var-loading\\">
  <div class=\\"var-loading__content var-loading__content--active\\">
    <div>test</div>
  </div>
  <div class=\\"var--box var-loading__body var-loading__inside\\">
    <div class=\\"var-loading__circle\\"><span class=\\"var-loading__circle-block\\" style=\\"width: 30px; height: 30px; color: currentColor;\\"><svg viewBox=\\"25 25 50 50\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\"></circle></svg></span></div>
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
    <div class=\\"var-loading__description var-loading__description--normal\\" style=\\"color: currentColor;\\">loading....</div>
  </div>
</div>"
`;

exports[`test wave props 1`] = `
"<div class=\\"var-loading\\">
  <!--v-if-->
  <div class=\\"var--box var-loading__body\\">
    <!--v-if-->
    <div class=\\"var-loading__wave var-loading__wave--small\\">
      <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__wave-item var-loading__wave-item--small\\"></div>
      <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__wave-item var-loading__wave-item--small\\"></div>
      <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__wave-item var-loading__wave-item--small\\"></div>
      <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__wave-item var-loading__wave-item--small\\"></div>
      <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__wave-item var-loading__wave-item--small\\"></div>
    </div>
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;
